qt/app.py

```diff
@@ -223,6 +231,10 @@ def show_details(self):
     def showResultsWindow(self):
         if self.resultWindow is not None:
             if self.use_tabs:
+                if self.main_window.indexOfWidget(self.resultWindow) < 0:
+                    self.main_window.addTab(
+                        self.resultWindow, "Results", switch=True)
+                    return
```

Comment:
Indeed. Ideally this would have been part of a separate PR, but these changes relied on changes from this particular branch (related to preference dialog window mostly), so I decided to keep there here. They are still in separate commits at least.